plugins/drivers/driver.go

```diff
@@ -87,14 +87,6 @@ type DriverNetworkManager interface {
 	DestroyNetwork(allocID string, spec *NetworkIsolationSpec) error
 }
 
-// InternalDriverPlugin is an interface that exposes functions that are only
-// implemented by internal driver plugins.
-type InternalDriverPlugin interface {
-	// Shutdown allows the plugin to cleanup any running state to avoid leaking
-	// resources. It should not block.
-	Shutdown()
```

Comment:
So effectively the win here is that by using a context injected by the caller instead of relying on callers to call `Shutdown` to the plugin, we effectively force the caller to handle the cancellation case?

Reply:
Yes.  Also, makes handling shutdown more explicit on the driver side, as the function isn't part of plugin interface.  A driver may accidentally not implement it, or expose a Shutdown function with the "wrong" signature (e.g. [1] though this is an external plugin example where previously shutdown wouldn't work anyway) without noticing.

[1] https://github.com/hashicorp/nomad-driver-ecs/blob/cc3a38f9d3411eb5c61d99e06106afd0210b85a6/ecs/driver.go#L218-L221